File: calculator.py
def read_number(prompt):
	while True:
		value = input(prompt).strip()
		if value.lower() == "q":
			return None
		try:
			return float(value)
		except ValueError:
			print("Please enter a valid number or q to quit.")


def main():
	print("Python Calculator")
	print("Enter q at any prompt to quit.")

	while True:
		first_number = read_number("Enter first number: ")
		if first_number is None:
			break

		operator = input("Enter an operator (+, -, *, /): ").strip()
		if operator.lower() == "q":
			break

		second_number = read_number("Enter second number: ")
		if second_number is None:
			break

		if operator == "+":
			result = first_number + second_number
		elif operator == "-":
			result = first_number - second_number
		elif operator == "*":
			result = first_number * second_number
		elif operator == "/":
			if second_number == 0:
				print("Cannot divide by zero.")
				continue
			result = first_number / second_number
		else:
			print("Unknown operator. Use +, -, *, or /.")
			continue

		print(f"Result: {result}")

	print("Calculator closed.")


if __name__ == "__main__":
	main()

File: calculator.java
import java.util.Scanner;

public class calculator {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Java Calculator");
        System.out.println("Enter q at any prompt to quit.");

        while (true) {
            System.out.print("Enter first number: ");
            if (!scanner.hasNextDouble()) {
                if (scanner.next().equalsIgnoreCase("q")) {
                    break;
                }
                System.out.println("Please enter a valid number.");
                continue;
            }
            double firstNumber = scanner.nextDouble();

            System.out.print("Enter an operator (+, -, *, /): ");
            String operator = scanner.next();
            if (operator.equalsIgnoreCase("q")) {
                break;
            }

            System.out.print("Enter second number: ");
            if (!scanner.hasNextDouble()) {
                if (scanner.next().equalsIgnoreCase("q")) {
                    break;
                }
                System.out.println("Please enter a valid number.");
                continue;
            }
            double secondNumber = scanner.nextDouble();
            double result;

            switch (operator) {
                case "+":
                    result = firstNumber + secondNumber;
                    break;
                case "-":
                    result = firstNumber - secondNumber;
                    break;
                case "*":
                    result = firstNumber * secondNumber;
                    break;
                case "/":
                    if (secondNumber == 0) {
                        System.out.println("Cannot divide by zero.");
                        continue;
                    }
                    result = firstNumber / secondNumber;
                    break;
                default:
                    System.out.println("Unknown operator. Use +, -, *, or /.");
                    continue;
            }

            System.out.println("Result: " + result);
        }

        scanner.close();
        System.out.println("Calculator closed.");
    }
}
